src/vivarium_gates_mncnh/components/antenatal_care.py

```diff
@@ -0,0 +1,292 @@
+        pop = self.population_view.get(index)
+        pop[COLUMNS.ATTENDED_CARE_FACILITY] = True
+        self.population_view.update(pop)
+
+
+class StandardUltrasound(TransientDecisionTreeState):
+    def __init__(self) -> None:
+        super().__init__("standard_ultasound")
+
+    @property
+    def columns_required(self) -> list[str]:
+        return [COLUMNS.ULTRASOUND_TYPE]
+
+    def transition_side_effect(self, index: pd.Index, _event_time: ClockTime) -> None:
+        pop = self.population_view.get(index)
+        pop[COLUMNS.ULTRASOUND_TYPE] = ULTRASOUND_TYPES.STANDARD
+        self.population_view.update(pop)
+
+
+class AIAssistedUltrasound(TransientDecisionTreeState):
+    def __init__(self) -> None:
+        super().__init__("ai_assisted_ultrasound")
+
+    @property
+    def columns_required(self) -> list[str]:
+        return [COLUMNS.ULTRASOUND_TYPE]
+
+    def transition_side_effect(self, index: pd.Index, _event_time: ClockTime) -> None:
+        pop = self.population_view.get(index)
+        pop[COLUMNS.ULTRASOUND_TYPE] = ULTRASOUND_TYPES.AI_ASSISTED
+        self.population_view.update(pop)
+
+
+def ANC() -> Machine:
+    initial_state = DecisionTreeState("initial")
+    attended_antental_care = ANCState()
+    gets_ultrasound = TransientDecisionTreeState("gets_ultrasound")
+    standard_ultasound = StandardUltrasound()
+    ai_assisted_ultrasound = AIAssistedUltrasound()
+    end_state = DecisionTreeState("end")
+
+    # Decisions
+    initial_state.add_decision(
+        attended_antental_care,
+        # TODO: this data will need to be updated when the ANC artifact is finished
+        lambda index: pd.Series(
+            ANC_RATES.ATTENDED_CARE_FACILITY[initial_state.location], index=index
+        ),
+    )
+    initial_state.add_decision(
+        end_state,
+        lambda index: pd.Series(
+            1 - ANC_RATES.ATTENDED_CARE_FACILITY[initial_state.location], index=index
+        ),
+    )
+    attended_antental_care.add_decision(
+        gets_ultrasound,
+        lambda index: pd.Series(
+            ANC_RATES.RECEIVED_ULTRASOUND[attended_antental_care.location], index=index
+        ),
+    )
+    attended_antental_care.add_decision(
+        end_state,
+        lambda index: pd.Series(
+            1 - ANC_RATES.RECEIVED_ULTRASOUND[attended_antental_care.location], index=index
+        ),
+    )
+    gets_ultrasound.add_decision(
+        standard_ultasound,
+        lambda index: pd.Series(
+            ANC_RATES.ULTRASOUND_TYPE[ULTRASOUND_TYPES.STANDARD], index=index
+        ),
+    )
+    gets_ultrasound.add_decision(
+        ai_assisted_ultrasound,
+        lambda index: pd.Series(
+            1 - ANC_RATES.ULTRASOUND_TYPE[ULTRASOUND_TYPES.AI_ASSISTED], index=index
+        ),
+    )
+    standard_ultasound.add_decision(end_state, lambda index: pd.Series(1.0, index=index))
+    ai_assisted_ultrasound.add_decision(end_state, lambda index: pd.Series(1.0, index=index))
```

Comment:
If you make this lambda function the default value for the probability function argument of `add_decision`, you can avoid having to specify it here like this.